toolchain/install/install_paths.cpp

```diff
@@ -76,6 +76,47 @@ auto InstallPaths::Make(llvm::StringRef install_prefix) -> InstallPaths {
   return paths;
 }
 
+auto InstallPaths::FindPreludeFiles() const
+    -> ErrorOr<llvm::SmallVector<std::string>> {
+  // This is structured to avoid a vector copy on success.
+  ErrorOr<llvm::SmallVector<std::string>> result =
+      llvm::SmallVector<std::string>();
+
+  std::string dir = core_package();
+
+  // Include <data>/core/prelude.carbon, which is the entry point into the
+  // prelude.
+  {
+    llvm::SmallString<256> prelude_file(dir);
+    llvm::sys::path::append(prelude_file, llvm::sys::path::Style::posix,
+                            "prelude.carbon");
+    result->push_back(prelude_file.str().str());
+  }
+
+  // Glob for <data>/core/prelude/**/*.carbon and add all the files we find.
+  // TODO: This should be using the vfs instead of actual filesystem.
```

Comment:
Does this TODO still make sense? `InstallPaths`, as it stands, doesn't make use of the VFS at all, so if this functionality fits here (which I think it does) it seems logical that it wouldn't do so either.
```suggestion
```
Perhaps a good model here would be that (if needed) we construct a fake `InstallPaths` for testing, for which we specify a list of prelude files on construction. But for this PR, I think what you have here is fine.

Reply:
Let me try to give an example of why I think this should be a TODO (you say "still", but I'm trying to add it).

We allow passing a vfs into the driver. Presumably we want to support, for example, people running the toolchain completely in-memory. However, because InstallPaths requires using the real filesystem, it means that some things will use the vfs, and some things won't.

For example, the toolchain is detected at a real filesystem path. That is used to get prelude paths. The prelude paths are read from the vfs, which consequently must be identical to the real filesystem paths.

A different approach, of course, would be to consistently use the real filesystem for things that come from `InstallPaths`. I suspect though that we would want to go in the other direction. Maybe I'm misunderstanding how others lean for vfs use?

Reply:
My thinking is that, while it seems good to have a TODO, it doesn't belong *here*. This class is all about finding things in the real file system. But in [testing/base/source_gen_test.cpp](https://github.com/carbon-language/carbon-lang/pull/4268/files#diff-7b8e6f9beadffd5a28d8b509c5e531f6808c9d90b5718af1b72b8ceaf5d25577), in `TestCompile`, we're passing an `InstallPaths` object that describes paths relative to the real file system (as all `InstallPaths` objects do today) and a VFS to the `Driver` constructor, and I think *that* is the thing that's wrong / should have a TODO, because that's where we're mixing up real file system and VFS stuff. So I'd be happier if the TODO lived there rather than here. But as noted, I'm happy for you to merge this as-is, with or without the TODO here.

As a fix for the TODO, I think a clean approach here would be to make `InstallPaths` more of an abstraction with a "real file system" implementation (this one), and a "fake file system for tests" implementation that uses the list of prelude file names that it copied into the VFS as the result of its `FindPreludeFiles` call. (But ideally not in this PR!)

Reply:
Moved the comment to the class level in the header, does that make sense? (in part to keep it in a central spot).

Note, I think we may want to support VFS for non-test Driver invocations.